### PR TITLE
Converts some js to tsx

### PR DESCRIPTION
**Pull request by dragomagol**
## About The Pull Request
Dev goals for changing js TGUI files to tsx. For maintainability.
Also removes some unused variables from the smoke machine.

## Changelog
Nothing player-facing

### CODE REVIEW

File: code/modules/reagents/chemistry/machinery/smoke_machine.dm [hunk 1]
```diff
@@ -126,18 +125,16 @@
 
 /obj/machinery/smoke_machine/ui_data(mob/user)
 	var/data = list()
-	var/TankContents[0]
-	var/TankCurrentVolume = 0
+	var/tank_contents[0]
```
Comment (ZephyrTFA) [inline code/modules/reagents/chemistry/machinery/smoke_machine.dm:128]: please don't define lists like this

File: code/modules/reagents/chemistry/machinery/smoke_machine.dm [hunk 2]
```diff
@@ -126,18 +125,16 @@
...
+	var/tank_current_volume = 0
 	for(var/datum/reagent/R in reagents.reagent_list)
-		TankContents.Add(list(list("name" = R.name, "volume" = R.volume))) // list in a list because Byond merges the first list...
-		TankCurrentVolume += R.volume
-	data["TankContents"] = TankContents
-	data["isTankLoaded"] = reagents.total_volume ? TRUE : FALSE
-	data["TankCurrentVolume"] = reagents.total_volume ? reagents.total_volume : null
-	data["TankMaxVolume"] = reagents.maximum_volume
+		tank_contents.Add(list(list("name" = R.name, "volume" = R.volume))) // list in a list because Byond merges the first list...
```
Comment (ZephyrTFA) [inline code/modules/reagents/chemistry/machinery/smoke_machine.dm:131]: can this be +=

File: tgui/packages/tgui/interfaces/BrigTimer.tsx [hunk 3]
```diff
@@ -2,8 +2,16 @@ import { useBackend } from '../backend';
 import { Button, Section } from '../components';
 import { Window } from '../layouts';
 
+type BrigTimerContext = {
+  timing: boolean;
```
Comment (ZephyrTFA) [inline tgui/packages/tgui/interfaces/BrigTimer.tsx:6]: boolean like

File: tgui/packages/tgui/interfaces/BrigTimer.tsx [hunk 4]
```diff
@@ -2,8 +2,16 @@ import { useBackend } from '../backend';
...
+  minutes: number;
+  seconds: number;
+  flash_charging: boolean;
```
Comment (ZephyrTFA) [inline tgui/packages/tgui/interfaces/BrigTimer.tsx:9]: see above

File: tgui/packages/tgui/interfaces/Clipboard.tsx [hunk 5]
```diff
@@ -2,8 +2,17 @@ import { useBackend } from '../backend';
 import { Box, Button, Divider, LabeledList, Flex, Section } from '../components';
 import { Window } from '../layouts';
 
+type ClipboardContext = {
+  pen: string;
+  integrated_pen: boolean;
```
Comment (ZephyrTFA) [inline tgui/packages/tgui/interfaces/Clipboard.tsx:7]: boolean like

File: tgui/packages/tgui/interfaces/EightBallVote.tsx [hunk 6]
```diff
@@ -3,8 +3,20 @@ import { Box, Button, Grid, Section, NoticeBox } from '../components';
 import { toTitleCase } from 'common/string';
 import { Window } from '../layouts';
 
+type EightBallVoteContext = {
+  shaking: boolean;
```
Comment (ZephyrTFA) [inline tgui/packages/tgui/interfaces/EightBallVote.tsx:7]: bl

File: tgui/packages/tgui/interfaces/EightBallVote.tsx [hunk 7]
```diff
@@ -3,8 +3,20 @@ import { Box, Button, Grid, Section, NoticeBox } from '../components';
...
+  question: string;
+  answers: Answer[];
+};
+
+type Answer = {
+  answer: string;
+  amount: number;
+  selected: boolean;
```
Comment (ZephyrTFA) [inline tgui/packages/tgui/interfaces/EightBallVote.tsx:15]: bl

File: tgui/packages/tgui/interfaces/SmokeMachine.tsx [hunk 8]
```diff
@@ -2,16 +2,28 @@ import { useBackend } from '../backend';
 import { AnimatedNumber, Box, Button, LabeledList, ProgressBar, Section } from '../components';
 import { Window } from '../layouts';
 
+type SmokeMachineContext = {
+  tankContents: Reagent[];
+  tankCurrentVolume: number;
+  tankMaxVolume: number;
+  active: boolean;
```
Comment (ZephyrTFA) [inline tgui/packages/tgui/interfaces/SmokeMachine.tsx:9]: .

**Review by ZephyrTFA — DISMISSED** [review]

File: tgui/packages/tgui/interfaces/BlackMarketUplink.tsx [hunk 9]
```diff
@@ -3,8 +3,39 @@ import { AnimatedNumber, Box, Button, Modal, Section, Stack, Tabs } from '../com
...
 
+type BlackMarketUplinkContext = {
+  categories: string[];
+  markets: Market[];
+  items: Item[];
+  money: number;
+  viewing_market: string;
+  viewing_category: string;
+  buying: boolean;
+  ltsrbt_built: boolean;
```
Comment (MrMelbert) [inline tgui/packages/tgui/interfaces/BlackMarketUplink.tsx:14]: I believe we use `BooleanLike` when handling booleans passed in from byond (which are `1` or `0`, and not `true` or `false`)

File: tgui/packages/tgui/interfaces/BlackMarketUplink.tsx [hunk 10]
```diff
@@ -93,8 +124,16 @@ export const BlackMarketUplink = (props, context) => {
   );
 };
 
+type ShipmentSelectorContext = {
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/BlackMarketUplink.tsx:127]: no need to differentiate these really

File: tgui/packages/tgui/interfaces/BlackMarketUplink.tsx [hunk 11]
```diff
@@ -3,8 +3,39 @@ import { AnimatedNumber, Box, Button, Modal, Section, Stack, Tabs } from '../com
 import { formatMoney } from '../format';
 import { Window } from '../layouts';
 
+type BlackMarketUplinkContext = {
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/BlackMarketUplink.tsx:6]: there's only ever going to be one backend data sent here so it doesn't need to be wordier than "data". same for the other files

File: tgui/packages/tgui/interfaces/BlackMarketUplink.tsx (hunk 9 above)
Comment (dragomagol) [reply tgui/packages/tgui/interfaces/BlackMarketUplink.tsx:14]: FUCK

File: tgui/packages/tgui/interfaces/BlackMarketUplink.tsx (hunk 11 above)
Comment (stylemistake) [reply tgui/packages/tgui/interfaces/BlackMarketUplink.tsx:6]: Same, just use the name `Data` for all of them.

**Pull request merged**